### ISSUE TITLE
Remove --experimental-emit-debug-metadata flag from frontend_server/tool when possible

### ISSUE THREAD

**Issue by annagrin** [flutter/flutter#59902]
Remove --experimenal-emit-debug-metadata flag from frontend_server/tool once metadata  is no longer experimental and is always produced by the frontend server.

**Comment by christopherfujino** [comment]
@johnniwinther do you know what this flag is, and whether removing it is blocked?

**Comment by johnniwinther** [comment]
I'm not aware of it.

@jensjoha Do you know anything about this?

**Comment by jensjoha** [comment]
I don't even know what it is.

**Comment by christopherfujino** [comment]
This still defaults to false, so I'm going to leave this bug open: https://github.com/dart-lang/sdk/blob/main/pkg/frontend_server/lib/frontend_server.dart#L211

**Comment by matanlurey** [comment]
The CFE team will need to help us, I don't understand why Flutter always needs this enabled:

https://github.com/flutter/flutter/blob/e212d3dcd89f576fc8d85afcccd980cff38ea240/packages/flutter_tools/lib/src/compile.dart#L852

... but it [defaults to `false` in the CFE](https://github.com/dart-lang/sdk/blob/37bce89a04e638a93b2204a5dc8426612597556c/pkg/frontend_server/lib/frontend_server.dart#L218-L220).

**Comment by nshahan** [comment]
This argument gets passed through to DDC when it is invoked by the frontend server. It is needed to support debugging and expression compilation in web builds.

I think it is likely we can pass this flag to DDC by default.

@srujzs
Have you come across any use case where we don't want the debug metadata?

cc @Markzipan

**Comment by srujzs** [comment]
> Have you come across any use case where we don't want the debug metadata?

Nope, and at this point, we're long past the `experimental` stage. Having it emit metadata always SGTM.

**Comment by nshahan** [comment]
We still need the a flag to signal that we want to DDC/frontend server to output the metadata file. We could probably remove the "experimental" part of the flag name but I'd want to document that there is no guarantee or API one could depend on in the output file. We still need the ability to freely change the format as needed.
